bitnami/kafka/templates/broker/statefulset.yaml

```diff
@@ -0,0 +1,441 @@
+            - name: KAFKA_CLIENT_USERS
+              value: {{ join "," .Values.sasl.client.users | quote }}
+            - name: KAFKA_CLIENT_PASSWORDS
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "kafka.saslSecretName" . }}
+                  key: client-passwords
+            {{- end }}
+            {{- if regexFind "SASL" (upper .Values.listeners.interbroker.protocol) }}
+            - name: KAFKA_INTER_BROKER_USER
+              value: {{ .Values.sasl.interbroker.user | quote }}
+            - name: KAFKA_INTER_BROKER_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "kafka.saslSecretName" . }}
+                  key: inter-broker-password
+            {{- end }}
+            {{- if and .Values.kraft.enabled (regexFind "SASL" (upper .Values.listeners.controller.protocol)) }}
+            - name: KAFKA_CONTROLLER_USER
+              value: {{ .Values.sasl.controller.user | quote }}
+            - name: KAFKA_CONTROLLER_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "kafka.saslSecretName" . }}
+                  key: controller-password
+            {{- end }}
+            {{- end }}
+            {{- if .Values.metrics.jmx.enabled }}
+            - name: JMX_PORT
+              value: "5555"
+            {{- end }}
+            {{- if .Values.broker.extraEnvVars }}
+            {{- include "common.tplvalues.render" ( dict "value" .Values.broker.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+            {{- if .Values.extraEnvVars }}
+            {{- include "common.tplvalues.render" ( dict "value" .Values.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          {{- if or .Values.broker.extraEnvVarsCM .Values.extraEnvVarsCM .Values.broker.extraEnvVarsSecret .Values.extraEnvVarsSecret }}
+          envFrom:
+            {{- if .Values.broker.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.broker.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.broker.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.broker.extraEnvVarsSecret "context" $) }}
+            {{- end }}
+            {{- if .Values.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.extraEnvVarsSecret "context" $) }}
+            {{- end }}
+          {{- end }}
+          ports:
+            - name: client
+              containerPort: {{ .Values.listeners.client.port }}
```

Comment:
Could you explain why this one is not following the `containerPorts` model?

Reply:
Because I wanted to model all Kafka settings under `listeners`, in this case, `listeners.<listener_name>.port`.
Adding `containerPorts.*`, would mean removing `listeners.<listener_name>.port` as it would not make sense having to configure both.

Reply:
The thing is, this would force to have the same port in the service and the container, is that correct?

Reply:
No no, `service.ports` still applies, but `containerPorts.client` would be replaced by `listeners.client.port`.

I thought it may be more intuitive for users to have all listener-related settings under a common object instead of spread among different places `containerPorts`, `auth.client.protocol`, `tls.client.sslClientAuth`, `clientListenerName`, etc.:
```
listeners:
  client:
    port: 9092
    protocol: SASL_PLAINTEXT
    name: CLIENT
    sslClientAuth: ""
```
This common configuration interface is also used to configure the `listeners.extraListeners`, see for example:
```
{{- $listeners := list .Values.listeners.client .Values.listeners.interbroker -}}
{{- range $i := .Values.listeners.extraListeners -}}
{{- $listeners = append $listeners $i -}}
{{- end -}}
...
{{- range $listener := $listeners -}}
{{- if regexFind "SSL" (upper $listener.protocol) -}}
...
{{- end -}}
{{- end -}}
```

Reply:
I see the point, but it's true that it is not directly intuitive that listener.client.port would be mapped to the container port and not the service port. Would it make sense to at least call it `listener.client.containerPort` so it is more closely related to the standards we have in the other charts? I'm concerned of the deviation this may cause.

Reply:
Sure, `listener.client.containerPort` sounds good to me